[AArch64][llvm] Armv9.7-A: Add support for FEAT_CMH and FEAT_LSCP

jthackray opened the pull request:
Add the following instructions to support:
   * `FEAT_CMH`: Contention Management Hints extension
       * `SHUH` instruction

   * `FEAT_LSCP`: Load-acquire and store-release pair extension
       * `STLP` instruction
       * `LDAP` instruction
       * `LDAPP` instruction

and system registers:
   - `TLBIDIDR_EL1`
   - `VTLBID<n>_EL2`
   - `VTLBIDOS<n>_EL2`

as documented here:

  * https://developer.arm.com/documentation/109697/2025_09/2025-Architecture-Extensions
  * https://developer.arm.com/documentation/ddi0602/2025-09/

Co-authored-by: Martin Wehking <martin.wehking@arm.com>

jthackray committed:
[AArch64][llvm] Armv9.7-A: Add support for FEAT_CMH and FEAT_LSCP

Add the following instructions to support:
   * `FEAT_CMH`: Contention Management Hints extension
       * `SHUH` instruction

   * `FEAT_LSCP`: Load-acquire and store-release pair extension
       * `STLP` instruction
       * `LDAP` instruction
       * `LDAPP` instruction

as documented here:

  * https://developer.arm.com/documentation/109697/2025_09/2025-Architecture-Extensions
  * https://developer.arm.com/documentation/ddi0602/2025-09/

Co-authored-by: Martin Wehking <martin.wehking@arm.com>

CarolineConcatto reviewed (approved):
LGTM, just address the nit before merging

jthackray committed:
fixup! [AArch64][llvm] Armv9.7-A: Add support for FEAT_CMH and FEAT_LSCP

Oops, forgot to upstream the testcases, and so found I'd missed a bet0 patch
from Martin. D'oh!

jthackray committed:
fixup! [AArch64][llvm] Armv9.7-A: Add support for FEAT_CMH and FEAT_LSCP

Small fix to test script

jthackray commented:
### Merge activity

* **Oct 23, 12:32 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/163155).
